src/nbl/asset/utils/CJITIncludeLoader.cpp

```diff
@@ -0,0 +1,61 @@
+
+#include "CJITIncludeLoader.h"
+
+
+CJITIncludeLoader::CJITIncludeLoader()
+{
+    m_includes["device_capabilities.hlsl"] = collectDeviceCaps();
+}
+
+std::string CJITIncludeLoader::loadInclude(const std::string path)
+{
+    const std::string prefix = "nbl/builtin/hlsl/jit/";
+    if (path.compare(0, prefix.size(), prefix) == 0)
+    {
+        std::string includeFileName = path.substr(prefix.size());
+
+        // Look up the content in m_includes map
+        auto it = m_includes.find(includeFileName);
+        if (it != m_includes.end())
+        {
+            // Return the content of the specified include file
+            return it->second;
+        }
+        else
+        {
+            // Handle error: include file not found
+            std::cerr << "Error: Include file '" << path << "' not found!" << std::endl;
+            return "";
+        }
+    }
+    else
+    {
+        // Handle error: invalid include path
+        std::cerr << "Error: Invalid include path '" << path << "'!" << std::endl;
+        return "";
+    }
```

Comment:
don't do any error handling/reporting, the function you;re suppsoed to override `IShadercompiler::IIncludeLoader::getInclude` returns an `std::optional<string>` for a reason

return an empty optional if you can't find/match anything